### PR TITLE
[ROCm] add ROCm support for XLA RCCL thunk

### PR DESCRIPTION
**Pull request by jeffdaily**

### CODE REVIEW

File: tensorflow/compiler/xla/service/gpu/BUILD [hunk 1]
```diff
@@ -399,9 +401,20 @@ filegroup(
     ),
 )
 
+# virtual targets since nested select statements not possible
+tf_cuda_library(
```
Comment (chsigg) [inline tensorflow/compiler/xla/service/gpu/BUILD]: Would an alias with an empty stub work instead?

Comment (jeffdaily) [reply tensorflow/compiler/xla/service/gpu/BUILD]: The code here was mostly copied from the RCCL support in `tensorflow/core/kernels/BUILD`, see https://github.com/tensorflow/tensorflow/blob/2e98e890918e0b51727b3d76edd72922bfe7184b/tensorflow/core/kernels/BUILD#L202-L236 .

I'm not familiar with the alias and empty stub approach.  Do you have sample code, or a doc link?

Comment (chsigg) [reply tensorflow/compiler/xla/service/gpu/BUILD]: Something like this?
```
cc_library(name = "empty")
alias (
  name = "virtual_nccl",
  actual = if_cuda("@local_config_nccl//:nccl", ":empty"),
)
alias (
  name = "virtual_rccl",
  actual = if_rocm("@local_config_rocm//rocm:rccl", ":empty"),
)
```

Comment (jeffdaily) [reply tensorflow/compiler/xla/service/gpu/BUILD]: Thank you for the example.  Pushed the change.

**Comment by deven-amd** [comment]
gentle ping

**Comment by deven-amd** [comment]
gentle ping

**Review by cheshire — APPROVED** [review]

**Pull request merged**